manifests/service.pp

```diff
@@ -18,19 +18,34 @@
 #   Whether the service should be managed at all.
 #   Defaults to true
 #
+# @param modular_services
+#   Hash of `service` resources for [modular drivers and sockets](https://libvirt.org/daemons.html#modular-driver-daemons).
+#   When this is set, `service_name`, `service_ensure`, and `service_enable` are ignored.
+#
 class libvirt::service (
   String  $service_name   = $libvirt::service_name,
   String  $service_ensure = $libvirt::service_ensure,
   Boolean $service_enable = $libvirt::service_enable,
   Boolean $manage_service = $libvirt::manage_service,
+  Hash[String, Hash] $modular_services = {},
 ) inherits libvirt {
   if $manage_service {
-    Libvirtd_conf <| |> ~> Service['libvirtd']
+    if $modular_services.empty {
+      Libvirtd_conf <| |> ~> Service['libvirtd']
```

Comment:
I think we should use tags to define what gets notified.
so: Libvirtd_conf <| |> ~> Service<| tag == 'libvirt-libvirtd-conf |>
So we can configure in hiera what to notify.